blockmanager.go

```diff
@@ -309,8 +327,18 @@ func (b *blockManager) isSyncCandidate(sp *serverPeer) bool {
 			return false
 		}
 	} else {
-		// The peer is not a candidate for sync if it's not a full node.
-		if sp.Services()&wire.SFNodeNetwork != wire.SFNodeNetwork {
+		// The peer is not a candidate for sync if it's not a full
+		// node. Additionally, if the segwit soft-fork package has
+		// activated, then the peer must also be upgraded.
+		segwitState, err := b.chain.ThresholdState(chaincfg.DeploymentSegwit)
```

Comment:
Introduced the new function to the `blockchain` package and replaced instances like the one above with a call to the function.